src/hooks/useFetch.js

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from 'react';
+
+const useFetch = (url, lat, lng) => {
+  const [data, setData] = useState([]);
+  const [status, setStatus] = useState('idle');
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (!url) return;
+
+    const getWikiEntries = async (lat, lng) => {
+      setStatus('loading');
+      try {
+        const res = await fetch(
+          `https://segdeha.com/api/nearby.php?lat=${lat}&lng=${lng}`,
```

Comment:
Similar to @laurenmbeatty’s suggestion about keeping your `lat` and `lng` in just one place, could we do the same with this URL? What I’d suggest is having this be the only component that ever makes API calls, so line 13 in `SearchResults.js` could lose the 3rd condition (i.e. you only need a valid `lat` and `lng` to make the call. What do you think?

Reply:
I’ve removed the hardcoded url from the useFetch hook. Would that solve the duplication problem? If no, I could try your suggestion and remove the 3rd condition on line 3 in SearchResults.js, it's just that I’m not sure I’ve understood how to implement it correctly😄